### PR TITLE
Column visibility constraints in patient view mutation table

### PR DESCRIPTION
**Pull request by adamabeshouse**
(1) Hide mRNA Expr rank if no expression porfile or more than one sample
(2) Hide Tumors if only one sample
(3) Hide copy number if no discrete cna profile

# Checks
- [x] Follows [7 rules of great commit messages](http://chris.beams.io/posts/git-commit/). For most PRs a single commit should suffice, in some cases multiple topical commits can be useful. During review it is ok to see tiny commits (e.g. Fix reviewer comments), but right before the code gets merged to master or rc branch, any such commits should be squashed since they are useless to the other developers. Definitely avoid [merge commits, use rebase instead.](http://nathanleclaire.com/blog/2014/09/14/dont-be-scared-of-git-rebase/)
- [x] Follows the [Airbnb React/JSX Style guide](https://github.com/airbnb/javascript/tree/master/react).
- [x] Make sure your commit messages end with a Signed-off-by string (this line
  can be automatically added by git if you run the `git-commit` command with
  the `-s` option)

### CODE REVIEW

**Comment by adamabeshouse** [comment]
[(3) Mutations Table --> Hide mRNA column if there is no expression profile.](https://trello.com/c/FPrzMRnG/133-3-mutations-table-hide-mrna-column-if-there-is-no-expression-profile)

File: src/pages/patientView/mutation/PatientViewMutationTable.tsx [hunk 1]
```diff
@@ -427,15 +449,30 @@ export default class PatientViewMutationTable extends React.Component<PatientVie
...
+            if (next === MutationTableColumn.MRNA_EXPR &&
+                (!this.props.mrnaExprRankGeneticProfileId
+                || this.getSamples().length > 1)) {
+                shouldAdd = false;
+            } else if (next === MutationTableColumn.TUMORS && this.getSamples().length < 2) {
+                shouldAdd = false;
+            } else if (next === MutationTableColumn.COPY_NUM && !this.props.discreteCNAGeneticProfileId) {
+                shouldAdd = false;
+            }
+            if (shouldAdd) {
```
Comment (onursumer) [inline src/pages/patientView/mutation/PatientViewMutationTable.tsx]: How about extending the default `Column` definition with and optional `shouldAdd` function and handle each column in a separate function? (If no `shouldAdd` provided, it should always add that column)

```JavaScript
MutationTableColumn<T> extends Column<T>
{
    shouldAdd: (...) => boolean;
}
...
...
if (column.shouldAdd(this.props.store)) {
    columns.push(...);
}
```

Comment (onursumer) [inline src/pages/patientView/mutation/PatientViewMutationTable.tsx]: Hmm, looks like the PR has changed, we don't have the `store` here anymore. Defining generic parameters for `shouldAdd` function is not trivial in this case :)

Comment (adamabeshouse) [inline src/pages/patientView/mutation/PatientViewMutationTable.tsx]: I went through the exact same thought process haha, I got stumped on what the arguments would be in the call signature

**Comment by adamabeshouse** [comment]
[bring over mutation table column order](https://trello.com/c/2aoVGBog/154-bring-over-mutation-table-column-order)

**Comment by adamabeshouse** [comment]
[BUG: Sort allele freq column fails](https://trello.com/c/igxnAFup/153-bug-sort-allele-freq-column-fails)

File: src/pages/patientView/mutation/PatientViewMutationTable.tsx [hunk 2]
```diff
@@ -175,7 +185,7 @@ function defaultFilter(data:Mutation[], dataField:string, filterString:string):b
 
 @observer
 export default class PatientViewMutationTable extends React.Component<PatientViewMutationTableProps,{}> {
-    @observable private _columns:{[columnEnum:number]:Column<Mutation[]>};
+    @observable private _columns:{[columnEnum:number]:Column<Mutation[]>&{order:number}};
```
Comment (onursumer) [inline src/pages/patientView/mutation/PatientViewMutationTable.tsx]: Define an extended type instead?
```JavaScript
MutationTableColumn<T> extends Column<T>
{
    order: number;
}
```

**Pull request merged**